core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -413,6 +414,34 @@ public void writeMap(@Nullable Map<String, Object> map) throws IOException {
     }
 
     /**
+     * write map to stream with consistent order
+     * to make sure every map generated bytes order are same
+     */
+    public void writeMapWithConsistentOrder(@Nullable Map<String, Object> map) throws IOException {
```

Comment:
Yes, If don't care about the value's type, can just use `writeGenericValue`.

About the **generic type** for **key**, I don't think we can do it, because if we use **generic type** for key, we need to use `writeGenericValue` for writing key. but for `writeGenericValue`, it will write extra byte for `Type`, like `String` use `byte 0` for `String` type. so if we do this, we need to create a new method maybe like `readGenericHashMap`.
but this will cause `writeMapWithConsistentOrder` method is incompatible with `readMap` and `readGenericValue`. so How about your idea?
 
and for the other `writeMap`, they also use `String` type as key. maybe we can transform `Integer` to String when we need to writeMap with consistent order.